File: backend/internal/dto/land_parcel.go
package dto

type ParcelQuery struct {
	Keyword  string
	State    string
	Page     int
	PageSize int
}

type CreateParcelRequest struct {
	ParcelCode       string `json:"parcel_code" validate:"required,min=2,max=64"`
	Name             string `json:"name" validate:"required,min=2,max=160"`
	BoundaryGeoJSON  string `json:"boundary_geojson" validate:"required"`
	CoordinateSystem string `json:"coordinate_system" validate:"required"`
	OwnerOrg         string `json:"owner_org" validate:"max=160"`
	ParcelState      string `json:"parcel_state" validate:"omitempty,oneof=active archived review"`
}

type UpdateParcelRequest struct {
	Name             *string `json:"name" validate:"omitempty,min=2,max=160"`
	BoundaryGeoJSON  *string `json:"boundary_geojson"`
	CoordinateSystem *string `json:"coordinate_system"`
	OwnerOrg         *string `json:"owner_org" validate:"omitempty,max=160"`
	ParcelState      *string `json:"parcel_state" validate:"omitempty,oneof=active archived review"`
	BoundaryVersion  *uint   `json:"boundary_version" validate:"omitempty,gt=0"`
}

File: frontend/src/api/land-parcel.ts
import { api } from './client'
import type { ApiEnvelope } from '@/types/api'
import type { LandParcel } from '@/types/land-parcel'

export interface ParcelQuery {
  keyword?: string
  state?: string
  page?: number
  page_size?: number
}

export type ParcelInput = {
  parcel_code?: string
  name?: string
  boundary_geojson?: string
  coordinate_system?: string
  owner_org?: string
  parcel_state?: string
  boundary_version?: number
}

export const landParcelApi = {
  list: (params?: ParcelQuery) => api.get<ApiEnvelope<LandParcel[]>>('/parcels', { params }),
  detail: (id: number) => api.get<ApiEnvelope<LandParcel>>(`/parcels/${id}`),
  create: (body: ParcelInput) => api.post<ApiEnvelope<LandParcel>>('/parcels', body),
  update: (id: number, body: ParcelInput) => api.patch<ApiEnvelope<LandParcel>>(`/parcels/${id}`, body),
}
